tests/test_average.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+import pytest
+import torch
+
+from tests.helpers.testers import MetricTester, NUM_BATCHES, BATCH_SIZE
+from torchmetrics.average import AverageMeter
+
+
+def average(values, weights):
+    return np.average(values, weights=weights)
+
+
+@pytest.mark.parametrize(
+    "values, weights",
+    [
+        (torch.rand(NUM_BATCHES, BATCH_SIZE), torch.ones(NUM_BATCHES, BATCH_SIZE)),
+        (torch.rand(NUM_BATCHES, BATCH_SIZE), torch.rand(NUM_BATCHES, BATCH_SIZE) > 0.5),
+        (torch.rand(NUM_BATCHES, BATCH_SIZE, 2), torch.rand(NUM_BATCHES, BATCH_SIZE, 2) > 0.5),
+    ],
+)
+class TestAverageMeter(MetricTester):
+    @pytest.mark.parametrize("ddp", [False, True])
+    @pytest.mark.parametrize("dist_sync_on_step", [False, True])
+    def test_average_fn(self, ddp, dist_sync_on_step, values, weights):
+        self.run_class_metric_test(
+            ddp=ddp,
+            dist_sync_on_step=dist_sync_on_step,
+            metric_class=AverageMeter,
+            sk_metric=average,
+            # Abuse of names here
+            preds=values,
+            target=weights,
```

Comment:
I wasn't really sure what to do here -- looking at the code, it looks like these are special-cased as positional arguments (which is why the differing names don't matter), and that we partition the data across processes using these.

Reply:
looks fine to me :]

Reply:
looks fine to me :]

Reply:
looks fine to me :]